core/src/main/scala/kafka/utils/Exit.scala

```diff
@@ -34,8 +34,8 @@ object Exit {
     throw new AssertionError("halt should not return, but it did.")
   }
 
-  def addShutdownHook(code: => Unit, name: Option[String] = None): Unit = {
-    JExit.addShutdownHook(() => code, name.orNull)
+  def addShutdownHook(statementByName: => Unit, name: Option[String] = None): Unit = {
```

Comment:
Scala is an expression oriented language, there are no statements. This could be simply `f` or `shutdownHook`. Also, I think `name` should not be optional and should be the first parameter.

Reply:
> there are no statements
> This could be simply `f` or `shutdownHook`

I was under the impression that statements were expressions that return no value (i.e. `:Unit`), but I could see how this may be technically incorrect.  I'll rename it `shutdownHook`.

> I think `name` should not be optional and should be the first parameter.

There are places where currently a name is not given, and the `Thread` class auto-generates a name when it is not supplied, but I don't see a problem with making it a mandatory first parameter and supplying a thread name in the places where one is not yet supplied to the shutdown hook thread; I will do that.  I will switch the order in the Java `Exit` class as well (`name` first, `Runnable` second).

Reply:
Yeah, `Unit` in Scala is actually a type and `()` a value. So, no statements. :)

With regards to names, I personally think it's very useful to give names to threads and we should require it. I think you've done that for the Scala code. Can we do it for the Java code too?